libcloudforensics/providers/gcp/forensics.py

```diff
@@ -227,3 +227,47 @@ def CreateDiskFromGCSImage(
       'md5Hash': md5_hash_hex
   }
   return result
+
+
+def InstanceNetworkQuarantine(project_id: str,
+                        instance_name: str,
+                        exempted_src_ips: Optional[List[str]] = None,
+                        enable_logging: bool = False) -> None:
+  """Put a Google Cloud instance in network quarantine.
+
+    Args:
+      project_id (str): Google Cloud Project ID.
+      instance_name (str): : The name of the virtual machine.
+      exempted_src_ips (List[str]): List of IPs exempted from the deny-all
+          ingress firewall rules, ex: analyst IPs.
+      enable_logging (bool): Optional. Enable firewall logging.
+          Default is False.
+  """
+  logger.info('Putting instance "{0:s}", in project {1:s}, in network '
+              'quarantine.'.format(instance_name, project_id))
+  project = gcp_project.GoogleCloudProject(project_id)
+  instance = project.compute.GetInstance(instance_name)
+  get_operation = instance.GetOperation()
+  network_interfaces = get_operation['networkInterfaces']
+  target_tags = []
+  for interface in network_interfaces:
+    network_name = interface["network"]
+    deny_ingress_tag = common.GenerateUniqueInstanceName(
+        'deny-ingress-tag-', common.COMPUTE_NAME_LIMIT)
+    deny_egress_tag = common.GenerateUniqueInstanceName(
+        'deny-egress-tag-', common.COMPUTE_NAME_LIMIT)
```

Comment:
Why do we need tags for here? Why Are we calling `GenerateUniqueInstanceName` if we're looking to generate a tag name?

Reply:
just to add a time stamp to the prefix. I used this to guarantee that the tag name will not collide with any other tag name, specially if there are multiple network interfaces, which means we have to add more tags to the same instance. I also used the same tag name as a firewall name later-on to guarantee it doesn't collide with other FW rule names